master/internal/api_trials.go

```diff
@@ -531,6 +534,41 @@ func (a *apiServer) PostTrialProfilerMetricsBatch(
 	return &apiv1.PostTrialProfilerMetricsBatchResponse{}, errs.ErrorOrNil()
 }
 
+func (a *apiServer) TrialPreemptionSignal(
+	req *apiv1.TrialPreemptionSignalRequest,
+	resp apiv1.Determined_TrialPreemptionSignalServer,
+) error {
+	eID, rID, err := a.m.db.TrialExperimentAndRequestID(int(req.TrialId))
+	switch {
+	case err == db.ErrNotFound:
+		return trialNotFound
+	case err != nil:
+		return err
+	}
+	trialAddr := actor.Addr("experiments", eID, rID)
+
+	signal := make(chan struct{}, 1)
+	a.m.system.TellAt(trialAddr, trialWatchPreemption{signal: signal})
+	unwatch := func() { a.m.system.TellAt(trialAddr, trialUnwatchPreemption{signal: signal}) }
+
+	for {
+		select {
+		case _, ok := <-signal:
+			if !ok {
+				// Do not call unwatch() when the channel is closed.
+				return nil
```

Comment:
So if the trial actor panics or shuts down for some reason, shouldn't we start returning True from the preemption API?

And if that is the case, is there a good reason to distinguish between a close message and just closing the channel?

Reply:
trial panicing is a good thought.. i'll think about it.

Reply:
if the trial closes i'm going to return to preempt.. anything still listening should definitely preempt